stellargraph/mapper/corrupted.py

```diff
@@ -96,31 +133,31 @@ def __getitem__(self, index):
 
         inputs, _ = self.base_generator[index]
 
-        if isinstance(
-            self.base_generator, (FullBatchSequence, SparseFullBatchSequence)
-        ):
+        shuffled_feats = []
+        for group in self.corrupt_index_groups:
+            feats_orig = [inputs[idx] for idx in group]
```

Comment:
I don't there's a check on `idx` anywhere, this could trigger a confusing warning for the user. 

We could either wrap this in an exception here:

```python
try:
    feats_orig = [inputs[idx] for idx in group]
except KeyError as e:
    raise KeyError("something about invalid index in corrupt_index_groups")
```

Or maybe do a check in `__init__`:

```python
peeked_inputs, _ = self.base_generator[0]
try:
    feats_orig = [[peeked_inputs[idx] for idx in group] for group in corrupt_index_groups]
except KeyError as e:
    raise KeyError("something about invalid index in corrupt_index_groups")
```

Reply:
I think the first one is better; the second one seems somewhat difficult to get "perfect", due to things like randomness (or other) problems with batch_num = 0 being requested twice, and handling of generators of length 0 (weird/useless, but still...).